mlflow/types/llm.py

```diff
@@ -58,6 +79,16 @@ def _convert_dataclass_list(self, key, cls, required=True):
     def to_dict(self):
         return asdict(self, dict_factory=lambda obj: {k: v for (k, v) in obj if v is not None})
 
+    @classmethod
+    def from_dict(self, data):
+        """
+        Create an instance of the class from a dict, ignoring any undefined fields.
+        This is useful when the dict contains extra fields, causing cls(**data) to fail.
+        """
+        field_names = [field.name for field in fields(self)]
+        filtered_data = {k: v for k, v in data.items() if k in field_names}
+        return self(**filtered_data)
```

Comment:
can we replace `self` with `cls` to follow the conventional naming in class methods?